cfgov/ask_cfpb/models/search/documents.py

```diff
@@ -0,0 +1,508 @@
+                sibling = self.answer_base.spanish_page
+            if sibling and sibling.live and not sibling.redirect_to_page:
+                return sibling.url
+
+    def get_context(self, request, *args, **kwargs):
+        portal_topic = self.primary_portal_topic or self.portal_topic.first()
+        context = super(AnswerPage, self).get_context(request)
+        context['related_questions'] = self.related_questions.all()
+        context['description'] = (
+            self.short_answer if self.short_answer
+            else truncate_preview(self.answer_content))
+        context['last_edited'] = self.last_edited
+        context['portal_page'] = get_portal_or_portal_search_page(
+            portal_topic, language=self.language)
+        context['breadcrumb_items'] = get_ask_breadcrumbs(
+            language=self.language,
+            portal_topic=portal_topic,
+        )
+        context['about_us'] = get_standard_text(self.language, 'about_us')
+        context['disclaimer'] = get_standard_text(self.language, 'disclaimer')
+        context['sibling_url'] = self.get_sibling_url()
+        return context
+
+    def answer_content_text(self):
+        raw_text = extract_raw_text(self.answer_content.stream_data)
+        return strip_tags(" ".join([self.short_answer, raw_text]))
+
+    def answer_content_data(self):
+        return truncate_preview(self.answer_content_text())
+
+    def short_answer_data(self):
+        return ' '.join(
+            RichTextField.get_searchable_content(self, self.short_answer))
+
+    def text(self):
+        short_answer = self.short_answer_data()
+        answer_text = self.answer_content_text()
+        full_text = f"{short_answer}\n\n{answer_text}\n\n{self.question}"
+        return full_text
+
+    def __str__(self):
+        if self.answer_base:
+            return f"{self.answer_base.id}: {self.title}"
+        else:
+            return self.title
+
+    @property
+    def clean_search_tags(self):
+        return [
+            tag.strip()
+            for tag in self.search_tags.split(",")
+        ]
+
+    @property
+    def status_string(self):
+        if self.redirect_to_page:
+            if not self.live:
+                return ("redirected but not live")
+            else:
+                return ("redirected")
+        else:
+            return super(AnswerPage, self).status_string
+
+    # Returns an image for the page's meta Open Graph tag
+    @property
+    def meta_image(self):
+        if self.social_sharing_image:
+            return self.social_sharing_image
+
+        if not self.category.exists():
+            return None
+
+        return self.category.first().category_image
+
+    # Overrides the default of page.id for comparing against split testing
+    # clusters. See: core.feature_flags.in_split_testing_cluster
+    @property
+    def split_test_id(self):
+        return self.answer_base.id
+
+
+@registry.register_document
+class AnswerPageSearchDocument(Document):
```

Comment:
I think @alexm118 had made the case that if this class is in `ask_cfpb.models.search`, we should be fine with calling it `AnswerPageDocument`, because the context is clearer.

Reply:
Renamed AnswerPageSearchDocument to AnswerPageDocument

Reply:
Renamed class to AnswerPageDocument